offline_imu_cam/plugin.cpp

```diff
@@ -60,34 +53,16 @@ class offline_imu_cam : public ILLIXR::threadloop {
         const sensor_types& sensor_datum = _m_sensor_data_it->second;
         ++_m_sensor_data_it;
 
-        imu_cam_log.log(record{imu_cam_record,
-                               {
-                                   {iteration_no},
-                                   {bool(sensor_datum.cam0)},
-                               }});
-
-        std::optional<cv::Mat> cam0 =
-            sensor_datum.cam0 ? std::make_optional<cv::Mat>(*(sensor_datum.cam0.value().load().release())) : std::nullopt;
-        RAC_ERRNO_MSG("offline_imu_cam after cam0");
-
-        std::optional<cv::Mat> cam1 =
-            sensor_datum.cam1 ? std::make_optional<cv::Mat>(*(sensor_datum.cam1.value().load().release())) : std::nullopt;
-        RAC_ERRNO_MSG("offline_imu_cam after cam1");
-
-#ifndef NDEBUG
-        /// If debugging, assert the image is grayscale
-        if (cam0.has_value() && cam1.has_value()) {
-            const int num_ch0 = cam0.value().channels();
-            const int num_ch1 = cam1.value().channels();
-            assert(num_ch0 == 1 && "Data from lazy_load_image should be grayscale");
-            assert(num_ch1 == 1 && "Data from lazy_load_image should be grayscale");
+        if (sensor_datum.cam0 && sensor_datum.cam1) {
+            cv::Mat cam0 = *(sensor_datum.cam0.value().load().release());
```

Comment:
Yes right. But I'm not going to include `offline_imu_cam`. Will remove it in the final version.

Reply:
But I thought we hadn't fully figured out the timing issue with separate IMU & CAM yet?

Reply:
True. But it would be weird if we keep `offline_imu`, `offline_cam`, and `offline_imu_cam`. Either we keep `offline_imu_cam` or `offline_imu` and `offline_cam`. I prefer the latter one, while we continue debugging.